### PR TITLE
Only send the opt-in ping if it actually changed

### PR DESCRIPTION
**Pull request by joshaber**
Noticed this while working on #1724.

### CODE REVIEW

**Comment by joshaber** [comment]
🏷 💣

File: app/src/lib/stats/stats-store.ts [hunk 1]
```diff
@@ -231,12 +231,16 @@ export class StatsStore {
   }
 
   /** Set whether the user has opted out of stats reporting. */
-  public setOptOut(optOut: boolean): Promise<void> {
+  public async setOptOut(optOut: boolean): Promise<void> {
+    const changed = this.optOut !== optOut
```
Comment (niik) [inline app/src/lib/stats/stats-store.ts]: Any reason why you chose not to just early return here?

**Comment by joshaber** [comment]
Can do 👍 Note that we can't completely short-circuit because in the case of the Welcome flow, the default value of opting it might be equal to their chosen value, but we still wanna set `localStorage` so indicate the user has made their choice.

File: app/src/lib/stats/stats-store.ts (hunk 1 above)
Comment (joshaber) [inline app/src/lib/stats/stats-store.ts]: Yup, I just commented below about that 😄

**Pull request merged**